_posts/2018-11-11-kanban.md

```diff
@@ -0,0 +1,124 @@
+
+![a screenshot of the tweet linked above](/log/assets/2018-11-11-kanban/tweet.png)
+
+Thanks due to some prodding by [@sigje](https://twitter.com/sigje/status/1025121668191023104) and some internal folks, I've finally come around to writing this up.
+
+![a screenshot of the tweet linked above](/log/assets/2018-11-11-kanban/prod.png)
+
+I'll start out with an overview of how the boards look like today, and will have the life of a ticket in the second half.
+
+# Dual-track Agile
+
+One of my big concerns last year was a big backlog of things we wanted to achieve with the PDK, but were unable to address at "this" point in time. With the default setup of JIRA (back then), this was loading up the board with tickets that were not relevant to the day-to-day activities. At the same time, this glut of tickets obscured the view of what should be coming next, as individual stories were too detailed, and not high-lighted.
+
+At the time I stumbled over some folks talking about dual-track agile: splitting design and long-term planning out from the day-to-day development board. This allowed me to represent the high-level planning and progress on one board, and to declutter the day-to-day detail view necessary for development on the primary board.
+
+# The "State of Play" Board
+
+The main board for day-to-day development usage should look familiar to everyone who had to work with JIRA.
+
+![the main board](/log/assets/2018-11-11-kanban/state_of_play_board.png)
+
+We were already used to using Epics to group tickets, so I followed JIRA's default of Epics as swimlanes. At that time we were using our internal JIRA defaults for board columns that was built for the much larger team and project of Puppet Enterprise, which had something like 8 different ticket states to handle Dev/QA/CI handoffs. In the PDK team we were mostly using [Travis CI](https://travis-ci.com) to directly test pull requests and code would go straight to master after a review from a second engineer. This resulted in the six columns that we have now: Backlog, Groomed, Blocked, In Progress, Review, and Done.
+
+Astute readers might notice that the backlog column doesn't show up on the screenshot. This is a recent change in JIRA that I really love. The first column of the board is not shown on the main board anymore, but moved to a separate - aptly named "backlog" - view.
+
+"Blocked" is a special case, mostly reserved for tickets we started, but now Need Information (or work) from some external party. This might be community feedback, or some service from another team. We've kept those tickets on the main board to not lose sight of them, and revisit them every so often to see if we need to follow-up.
```

Comment:
Maybe mention column limits, especially with regards to Blocked because when you exceed the column turns red.

This, however, may be too much information for your blog.  Just a thought.

Reply:
You're not the first one to bring up the question. I'll still leave it out, as this post is getting long already.

Reply:
Maybe do a part 2 :)